src/theory/evaluator.cpp

```diff
@@ -524,6 +552,460 @@ EvalResult Evaluator::evalInternal(TNode n,
+          const FloatingPoint& val = results[currNode[0]].d_fp;
+          results[currNode] = EvalResult(val.isNaN());
+          break;
+        }
+
+        case kind::FLOATINGPOINT_ISNEG:
+        {
+          const FloatingPoint& val = results[currNode[0]].d_fp;
+          results[currNode] = EvalResult(val.isNegative());
+          break;
+        }
+
+        case kind::FLOATINGPOINT_ISPOS:
+        {
+          const FloatingPoint& val = results[currNode[0]].d_fp;
+          results[currNode] = EvalResult(val.isPositive());
+          break;
+        }
+
+        case kind::FLOATINGPOINT_TO_FP_IEEE_BITVECTOR:
+        {
+          TNode op = currNode.getOperator();
+          const FloatingPointToFPIEEEBitVector& param =
+              op.getConst<FloatingPointToFPIEEEBitVector>();
+          const BitVector& val = results[currNode[0]].d_bv;
+          results[currNode] = EvalResult(
+              FloatingPoint(param.t.exponent(), param.t.significand(), val));
+          break;
+        }
+
+        case kind::FLOATINGPOINT_TO_FP_FLOATINGPOINT:
+        {
+          TNode op = currNode.getOperator();
+          const FloatingPointToFPFloatingPoint& param =
+              op.getConst<FloatingPointToFPFloatingPoint>();
+          const RoundingMode& rm = results[currNode[0]].d_rm;
+          const FloatingPoint& val = results[currNode[1]].d_fp;
+
+          results[currNode] = EvalResult(val.convert(param.t, rm));
+          break;
+        }
+
+        case kind::FLOATINGPOINT_TO_FP_REAL:
+        {
+          TNode op = currNode.getOperator();
+          const FloatingPointToFPFloatingPoint& param =
+              op.getConst<FloatingPointToFPFloatingPoint>();
+          const RoundingMode& rm = results[currNode[0]].d_rm;
+          const FloatingPoint& val = results[currNode[1]].d_fp;
+
+          results[currNode] = EvalResult(val.convert(param.t, rm));
+          break;
+        }
+
+        case kind::FLOATINGPOINT_TO_FP_SIGNED_BITVECTOR:
+        {
+          TNode op = currNode.getOperator();
+          const FloatingPointToFPFloatingPoint& param =
+              op.getConst<FloatingPointToFPSignedBitVector>();
+          const RoundingMode& rm = results[currNode[0]].d_rm;
+          const BitVector& val = results[currNode[1]].d_bv;
+
+          results[currNode] = EvalResult(FloatingPoint(param.t, rm, val, true));
+          break;
+        }
+
+        case kind::FLOATINGPOINT_TO_FP_UNSIGNED_BITVECTOR:
+        {
+          TNode op = currNode.getOperator();
+          const FloatingPointToFPFloatingPoint& param =
+              op.getConst<FloatingPointToFPUnsignedBitVector>();
+          const RoundingMode& rm = results[currNode[0]].d_rm;
+          const BitVector& val = results[currNode[1]].d_bv;
+
+          results[currNode] =
+              EvalResult(FloatingPoint(param.t, rm, val, false));
+          break;
+        }
+
+        case kind::FLOATINGPOINT_TO_UBV:
+        {
+          TNode op = currNode.getOperator();
```

Comment:
Also converted to _TOTAL version, as are other conversions from FP.

Reply:
Good to know, thanks!